js/src/Axis.ts

```diff
@@ -112,9 +118,9 @@ export class Axis extends WidgetView {
         } else if (num_ticks !== undefined && num_ticks !== null) {
             this.axis.tickValues(this.get_ticks_from_array_or_length());
         } else {
-            if (this.axis_scale.model.type === "ordinal") {
+            if (this.axis_scale instanceof OrdinalScale) {
                 this.axis.tickValues(this.axis_scale.scale.domain());
-            } else if (this.axis_scale.model.type === "date") {
+            } else if (this.axis_scale instanceof DateScale) {
```

Comment:
Should we get rid of the `type` attribute in each scale, if we now rely on the instance type?
https://github.com/davidbrochart/bqplot/blob/e4716a4e7a13f08a342359591a9a764aa7710543/js/src/DateScaleModel.ts#L40

Reply:
I think so yes. Although some bqplot plugins might still rely on them, so maybe we should show a deprecation warning for now.